### PR TITLE
Add GitHub Actions setting & fix `ocamyul.opam` to install dependecies

### PR DESCRIPTION
**Pull request by y-yu**
Example Result: https://github.com/y-yu/ocaml2evm/actions/runs/7667397901/job/20897133979?pr=1

```
Run opam exec -- dune exec ocamyulc sample/src/erc20.ml
/bin/sh: 1: solc: not found
Fatal error: exception Yojson__Common.Json_error("Blank input data")
```

Is `solc` not correct? 🤔 I think it should be `solcjs`.

### CODE REVIEW

**Comment by takumayoshioka** [comment]
We need to install `solc` via apt because `solcjs` is unable to compile Yul code, whereas `solc` can handle it. To install `solc`, please refer to https://docs.soliditylang.org/en/latest/installing-solidity.html#linux-packages .

**Comment by y-yu** [comment]
Thanks to your reply 🙏 

In [this GitHub Actions phase](https://github.com/y-yu/ocaml2evm/actions/runs/7667397901/job/20897133979?pr=1#step:4:1), I think that `solc` has been installed via running `npm install -g solc` command 🤔 
Could you tell me how to install `solc`?

**Comment by takumayoshioka** [comment]
The command-line executable of `solc` installed by npm is named `solcjs`, which is incompatible with `solc` installed by apt and does not have the option for compiling Yul codes. To install and use the `solc` command-line executable, please use the following commands:
```
sudo add-apt-repository ppa:ethereum/ethereum
sudo apt-get update
sudo apt-get install solc
```

**Comment by y-yu** [comment]
OK 👍  GitHub Actions success 🎉  since install `solc` using `apt-get` or `homebrew` (in macOS).

Example Result: https://github.com/y-yu/ocaml2evm/actions/runs/7685651558

File: ocamyul.opam [hunk 1]
```diff
@@ -1,4 +1,15 @@
 opam-version: "2.0"
+name: "ocaml2evm"
+version: "0.0.1"
```
Comment (yoshihiro503) [inline ocamyul.opam]:
```suggestion
version: "1.0.0"
```
I would like the version to be `1.0.0`.

**Comment by y-yu** [comment]
In this PR, I fixed `ocamyul.opam` more ad-hoc to install OPAM dependencies in GitHub Actions. Please tell me if you want to change something for `ocamyul.opam` 🙏

**Review by yoshihiro503 — COMMENTED** [review]
LGTM

File: ocamyul.opam (hunk 1 above)
Comment (y-yu) [reply ocamyul.opam]: OK, I fixed ✅

**Pull request merged**